runtime/j9vm/javanextvmi.cpp

```diff
@@ -763,15 +763,62 @@ JVM_IsImplicitlyConstructibleClass(JNIEnv *env, jclass cls)
 JNIEXPORT jboolean JNICALL
 JVM_IsNullRestrictedArray(JNIEnv *env, jobject obj)
 {
-	// TODO implement this with https://github.com/eclipse-openj9/openj9/issues/19460
-	return JNI_FALSE;
+	jboolean result = JNI_FALSE;
+	J9VMThread *currentThread = (J9VMThread *)env;
+	J9InternalVMFunctions *vmFuncs = currentThread->javaVM->internalVMFunctions;
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	if (NULL == obj) {
+		vmFuncs->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
+	} else {
+		jclass clazz = env->GetObjectClass(obj);
+		J9Class *j9clazz = J9VM_J9CLASS_FROM_JCLASS(currentThread, clazz);
+		if (J9_IS_J9ARRAYCLASS_NULL_RESTRICTED(j9clazz)) {
+			result = JNI_TRUE;
+		}
+	}
+	vmFuncs->internalExitVMToJNI(currentThread);
+	return result;
 }
 
 JNIEXPORT jarray JNICALL
-JVM_NewNullRestrictedArray(JNIEnv *env, jclass cls, jint length)
+JVM_NewNullRestrictedArray(JNIEnv *env, jclass componentType, jint length)
 {
-	assert(!"JVM_NewNullRestrictedArray unimplemented");
-	return NULL;
+	J9VMThread *currentThread = (J9VMThread *)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = currentThread->javaVM->internalVMFunctions;
+	J9Class *ramClass = NULL;
+	j9object_t newArray = NULL;
+	jarray arrayRef = NULL;
+
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	ramClass = J9VMJAVALANGCLASS_VMREF(currentThread, J9_JNI_UNWRAP_REFERENCE(componentType));
+
+	if (!(J9_IS_J9CLASS_VALUETYPE(ramClass) && J9_IS_J9CLASS_ALLOW_DEFAULT_VALUE(ramClass))) {
+		vmFuncs->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGILLEGALARGUMENTEXCEPTION, NULL);
+		goto done;
+	}
+
+	if (NULL == ramClass->nullRestrictedArrayClass) {
+		J9ROMArrayClass* arrayOfObjectsROMClass = (J9ROMArrayClass*)J9ROMIMAGEHEADER_FIRSTCLASS(vm->arrayROMClasses);
+		vmFuncs->internalCreateArrayClassWithOptions(currentThread, arrayOfObjectsROMClass, ramClass,
```

Comment:
https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#function-parameter-list-does-not-fit-on-one-line

Formatting for functions with params that exceed one line should be formatted like one of the two options in the document. However, note that even though that the examples in that document place the commas at the end of the lines, there is push back if they are not at the beginning of the next line.

Reply:
so the comma for ramClass would be like this?
```
		vmFuncs->internalCreateArrayClassWithOptions(
			currentThread, arrayOfObjectsROMClass, ramClass
			, J9_FINDCLASS_FLAG_CLASS_OPTION_NULL_RESTRICTED_ARRAY);
```

Reply:
I think the `J` should be on the same column as the above `c`.

Reply:
That is just what I have been told though. I personally think it is better with the comma on the preceding line.